src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -620,6 +622,14 @@ private void Dispose()
                 tid = results.Results.FindInt("thread-id");
             }
 
+            // Any existing variable objects at this point are from the last time we were in break mode, and are
+            //  therefore invalid.  Dispose them so they're marked for cleanup.
+            foreach (IVariableInformation varInfo in this.ActiveVariables)
```

Comment:
Might just be my bias as this is how VS does it, but it might make sense to do this when we continue instead of when we reenter break.